### ISSUE TITLE
Orb version 7.2.0 and 7.3.0 have not been published 😱

### ISSUE THREAD

**Issue by fotos** [react-native-community/react-native-circleci-orb#175]
### Orb version

`7.2.0` and `7.3.0` 

### What happened

The versions are [missing from the orb registry](https://circleci.com/developer/orbs/orb/react-native-community/react-native). The latest published version is `7.1.1`.

Additionally the last 2 releases are [missing from GitHub releases](https://github.com/react-native-community/react-native-circleci-orb/releases).

The [CircleCI pipeline failed with an error](https://app.circleci.com/pipelines/github/react-native-community/react-native-circleci-orb/339/workflows/fc35a7c6-8e9c-4171-ab21-d88bd47744f4/jobs/485):

```
[9:55:24 AM] [semantic-release] › ✖  Failed step "publish" of plugin "semantic-release-circleci-orb"
[9:55:24 AM] [semantic-release] › ✖  An error occurred while running semantic-release: Error: Command failed: /home/circleci/circleci orb publish packed-orb.yml react-native-community/react-native@7.2.0 --token [secure]
Error: AUTHORIZATION_FAILURE
```

Looks like the `token` is incorrect. A long time has passed since the last time the orb was published and I'm suspecting the token has been removed due to the [CircleCI security incident back in December 2022 / January 2023](https://circleci.com/blog/january-4-2023-security-alert/).

This probably requires manually (re-)running the CI steps to publish the orb and create the GH releases.

### Expected behavior

I'd like to use version `7.2.0`, which [makes the detox setup optional](https://github.com/react-native-community/react-native-circleci-orb/compare/v7.1.1...v7.2.0), in our project.

**Comment by fotos** [comment]
cc: @cortinico and @devnev who merged the last couple of PRs. 🙏

**Comment by cortinico** [comment]
I don't have rights to publish this orb, maybe @devnev has

**Comment by devnev** [comment]
Nope, I'm just a contributor. @matt-oakes seems to have set it up originally?

**Comment by fotos** [comment]
Hi @matt-oakes! 👋 

Is there anything I can do to help publish the missing orbs? Perhaps you can share the access with other maintainers to get this going?

**Comment by bhandanyan-nomad** [comment]
I'm trying to take advantage of the change introduced in this PR https://github.com/react-native-community/react-native-circleci-orb/pull/168 but it seems that it's only available in 7.3.0, which per this issue is not released.

@matt-oakes @devnev @cortinico if you could escalate to the right person to get this fixed it would be greatly appreciated 🙏

**Comment by cortinico** [comment]
I've sent an email to @matt-oakes as I believe he has the access to the orb publishing

**Comment by bhandanyan-nomad** [comment]
Any update here? If @matt-oakes can't be brought in to fix the issue, what else can be done?

For now, I'm working around my issue by pulling in the android docker image with my desired version manually without the rn/linux_android executor. It's a fine solution, but would still be nice to use what this orb provides.

Thanks 🙏

**Comment by matt-oakes** [comment]
Apologies for the lack of reply here. I don't routinely monitor this repo anymore as I don't use Circle CI.

I have replied to @cortinico's email and given them admin access to this repo. Additionally, there are already over 100 people with write access to this repo through the "react-native-community" members list.

It's not clear to me how to grant permission to publish on Circle CI, but @cortinico thinks that write/admin access to this Github repo will be enough. If it's not, then let me know and I can make the changes needed.

Again, apologies for the lack of reply.

**Comment by fotos** [comment]
Thank you @matt-oakes. I believe admin access is required in order to access the  "Security" section, in the "Settings" tab, and rotate (recreate) the CircleCI deploy keys.

@cortinico let me know if you need help with anything. For example, I believe, to re-publish the currently unpublished releases, we need to remove the [current](https://github.com/react-native-community/react-native-circleci-orb/releases/tag/v7.2.0) [tags ](https://github.com/react-native-community/react-native-circleci-orb/releases/tag/v7.3.0), and re-trigger the CI/CD that will create the GH release and publish the orbs.

**Comment by cortinico** [comment]
> @cortinico let me know if you need help with anything. For example, I believe, to re-publish the currently unpublished releases, we need to remove the [current](https://github.com/react-native-community/react-native-circleci-orb/releases/tag/v7.2.0) [tags ](https://github.com/react-native-community/react-native-circleci-orb/releases/tag/v7.3.0), and re-trigger the CI/CD that will create the GH release and publish the orbs.

Would you be up for fixing the Orb Development kit I started doing here?
- https://github.com/react-native-community/react-native-circleci-orb/pull/177

There should be some lint issues to fix and so on. Just send a similar PR and once the PR is green we can merge it and release a new version.

**Comment by fotos** [comment]
@cortinico sure, I can try.

I see that you did something similar to `circle orb init` in the repo. I think I can from here.

**Comment by fotos** [comment]
@cortinico as suspected, I don't think the current setup runs builds on forks: https://github.com/react-native-community/react-native-circleci-orb/pull/178

_(similarly with [other PRs](https://github.com/react-native-community/react-native-circleci-orb/pull/150) from [forks](https://github.com/react-native-community/react-native-circleci-orb/pull/172))_

Unfortunately I don't have write (push) access to the repo so I can't push changes directly in your branch:

```
➜  react-native-circleci-orb git:(nc/update-orb-development-kit) ✗ git push upstream nc/update-orb-development-kit
ERROR: Permission to react-native-community/react-native-circleci-orb.git denied to fotos.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I'm not sure I want to setup all the python tools locally (or in docker) to get this going. I think it's gonna be easier to get write access so I can push directly to the branch and iterate. 🙏

**Comment by cortinico** [comment]
> Unfortunately I don't have write (push) access to the repo so I can't push changes directly in your branch:

I've sent you an invite. Could you force push on the branch and make sure the CI passes so we can merge this?

**Comment by fotos** [comment]
Accepted the invite.

Will try to take a look over the weekend. Otherwise I’ll work on it during normal business hours next week.

**Comment by kevindice** [comment]
Thanks all - I too am keen on something available in 7.2.0.

**Comment by kevindice** [comment]
Hi @fotos, those of us using this orb are now seeing a fatal error in the `Configure Detox Environment` step which says that `brew tap` is not necessary to run anymore:

```
==> Tapping wix/brew
Cloning into '/opt/homebrew/Library/Taps/wix/homebrew-brew'...
remote: Enumerating objects: 747, done.        
remote: Counting objects: 100% (203/203), done.        
remote: Compressing objects: 100% (94/94), done.        
remote: Total 747 (delta 100), reused 188 (delta 91), pack-reused 544        
Receiving objects: 100% (747/747), 1.60 MiB | 2.44 MiB/s, done.
Resolving deltas: 100% (387/387), done.
Tapped 1 cask and 2 formulae (17 files, 1.6MB).
Error: Tapping homebrew/cask is no longer typically necessary.
Add --force if you are sure you need it done.

Exited with code exit status 1
```

In my case, I don't want to run this Detox config step at all.  We requested this to be optional in #167 and @devnev and @cortinico got #135 merged this summer.  It is included in the `7.2.0` version tag.  Other improvements are in `7.3.0`.

Without a fix to the publication pipeline for this package, we will be stuck on 7.1.1 indefinitely, and today I notice that 7.1.1 is now broken by this `brew tap` error.

Are you able to resolve the pipeline to get these versions published to Circle?  Or do you know who can?

**Comment by bhandanyan-nomad** [comment]
@fotos @cortinico @matt-oakes what's the latest on this issue? Can we expect it to be resolved anytime soon?

**Comment by cortinico** [comment]
The latest is that I'm looking for someone to help setting up the publishing of the orb. I have @fotos write access to the repo be he never followed up. If someone else is up for doing it, please raise your hand

**Comment by fotos** [comment]
Hey folks!

I was out on extended PTO for 3 weeks and now I'm back. I'll look at the issue again now.

Personally I'd rather avoid coupling the publishing of `7.2.0` and `7.3.0` to #177 (updating the development kit). If we can get `7.2.0` and `7.3.0` published by, for example, running the pipeline manually, that would be great. Then we could update the development kit and do another release.

In any case I'm gonna have a look now. If the upgrade is mandatory I'll try to open a PR by the end of the week. 

We also need the improvements from `7.2.0` and that's why I opened this issue.

**Comment by fotos** [comment]
There are 2 environment variables in the CircleCI configuration for `react-native-circleci-orb`:

<img width="940" alt="env-variables" src="https://github.com/react-native-community/react-native-circleci-orb/assets/11336/0e00b0dc-4ca5-4992-b307-076373a5af8a">

My recommendation is to rotate the `GH_TOKEN` token plus the `CIRCLECI_API_TOKEN` (which is used by https://github.com/matt-oakes/semantic-release-circleci-orb to publish the orb), and (I believe) that should be sufficient to publish new versions in the future.

For releasing `7.2.0` and `7.3.0`, after getting new values for the above environment variables, we can checkout the corresponding tags then [manually edit `.releaserc`](https://github.com/react-native-community/react-native-circleci-orb/blob/43c7903ccb0ad243dfefd306e1d4ead331c3f9b2/.releaserc#L21-L27) and trigger a `yarn semantic-release` to [publish the orb](https://github.com/react-native-community/react-native-circleci-orb/blob/43c7903ccb0ad243dfefd306e1d4ead331c3f9b2/.releaserc#L12-L19) and [create the GitHub releases](https://github.com/react-native-community/react-native-circleci-orb/blob/43c7903ccb0ad243dfefd306e1d4ead331c3f9b2/.releaserc#L20) (and [skip the `CHANGELOG`](https://github.com/react-native-community/react-native-circleci-orb/blob/43c7903ccb0ad243dfefd306e1d4ead331c3f9b2/.releaserc#L21-L27) which is already updated).

@cortinico I'm not sure why we have to update the SDK first – am I missing something?

I would gladly do the above, if I had permissions to create the tokens, which is going to unblock everyone. Then I can work to upgrade the CircleCI SDK so we can use the [SDK publishing process](https://circleci.com/docs/creating-orbs/) instead of the [manual publishing process](https://circleci.com/docs/orb-author-validate-publish/), independently and without pressure.

What do you think? Can I get ownership here and do the right thing?

**Comment by cortinico** [comment]
> Personally I'd rather avoid coupling the publishing of `7.2.0` and `7.3.0` to #177 (updating the development kit). If we can get `7.2.0` and `7.3.0` published by, for example, running the pipeline manually, that would be great. Then we could update the development kit and do another release.

I'm not too keen on releasing `7.2.0` and `7.3.0`. IMHO we can just publish a newer version from `main`.

> @cortinico I'm not sure why we have to update the SDK first – am I missing something?

The problem here is the `CIRCLECI_API_TOKEN` which we don't have practically (and I would refrain from using a personal token at this stage).
The new Orb guide is not requiring any API token at all (see https://circleci.com/docs/creating-orbs/) hence that's the reason why I think we should just update and release 7.4.0 (or 8.0.0) from main

**Comment by fotos** [comment]
> > Personally I'd rather avoid coupling the publishing of `7.2.0` and `7.3.0` to #177 (updating the development kit). If we can get `7.2.0` and `7.3.0` published by, for example, running the pipeline manually, that would be great. Then we could update the development kit and do another release.
> 
> I'm not too keen on releasing `7.2.0` and `7.3.0`. IMHO we can just publish a newer version from `main`.

Fair enough. 

_(We need to rename `master` to `main` – I might get this done after we get a new version out.)_

> 
> > @cortinico I'm not sure why we have to update the SDK first – am I missing something?
> 
> The problem here is the `CIRCLECI_API_TOKEN` which we don't have practically (and I would refrain from using a personal token at this stage). The new Orb guide is not requiring any API token at all (see https://circleci.com/docs/creating-orbs/) hence that's the reason why I think we should just update and release 7.4.0 (or 8.0.0) from main

Cool. I put a bit of effort to understand how the new SDK works (template, flow, scripts etc.) and got PR #177 working, including passing rudimentary linting / tests. I updated the PR description with more info. When you get a chance a chance have look and let's continue the discussion in the PR.

We are close to getting this issue resolve and having a new version published. 🎉

**Comment by cortinico** [comment]
> (We need to rename master to main – I might get this done after we get a new version out.)

We can do this whenever needed 👍

**Comment by fotos** [comment]
Great news, versions `7.2.0` and `7.3.0` are now published and available! 🎉 

* [7.2.0](https://circleci.com/developer/orbs/orb/react-native-community/react-native?version=7.2.0)
* [7.3.0](https://circleci.com/developer/orbs/orb/react-native-community/react-native?version=7.3.0)

I'm gonna continue working on getting the CI/CD pipeline working so we can merge pending PRs and release a subsequent version too.

_(If you are interested in the details of the whole saga there are available in #182)_

**Comment by fotos** [comment]
I got the GH releases published too:

* [v7.2.0](https://github.com/react-native-community/react-native-circleci-orb/releases/tag/v7.2.0)
* [v.7.3.0](https://github.com/react-native-community/react-native-circleci-orb/releases/tag/v7.3.0)
